xarray/backends/api.py

```diff
@@ -379,6 +380,18 @@ def _chunk_ds(
     return backend_ds._replace(variables)
 
 
+def _create_default_indexes(ds, create_default_indexes):
+    if not create_default_indexes:
```

Comment:
Could we move this check up in the stack? It looks weird inside the function here.

Reply:
I wouldn't mind having this check in multiple places. For the latter case you could use a lambda inside `pipe`. This seems more readable IMO than `_create_default_indexes(ds, create_default_indexes)`.

Reply:
it's `_maybe_create_default_indexes(ds, create_default_indexes)` now, so slightly better, but I do see your point. I'm honestly not enthusiastic about repeating the condition, so let me think about this until tomorrow

Reply:
Yeah I'm nit-picking. It's just that in general `do_something(..., do_something: bool)` feels odd to me.

I don't really see any issue with

```python
    if create_default_indexes:
        backend_ds = _maybe_create_default_indexes(backend_ds)
```

```python
    node.dataset.pipe(
        lambda ds: _maybe_create_default_indexes(ds) if create_default_indexes else ds
    ),
```